plugins/destination/csv/client/migrate.go

```diff
@@ -0,0 +1,78 @@
+package client
+
+import (
+	"context"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+func isSameColumns(record []string, columns []string) bool {
+	if len(record) != len(columns) {
+		return false
+	}
+	for i, v := range record {
+		if v != columns[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func (c *Client) migrate(tables schema.Tables) error {
+	for _, t := range tables {
+		filePath := path.Join(c.csvSpec.Directory, t.Name+".csv")
+		//nolint:gocritic,revive
+		if _, err := os.Stat(filePath); err == nil {
+			f, err := os.Open(filePath)
+			if err != nil {
+				return err
+			}
+			//nolint:revive
+			defer f.Close()
```

Comment:
I would wrap the whole block inside a fn (anonymous or a separate function) and remove the `nolint`. No reason to keep files open more than necessary.

Reply:
Good idea.